players/g1_player.py

```diff
@@ -286,36 +286,28 @@ def splash_zone_within_polygon(self, current_point: Tuple[float, float], target_
 
         distance = np.linalg.norm(np.array(current_point).astype(
             float) - np.array(target_point).astype(float))
+
         cx, cy = current_point
         tx, ty = target_point
         angle = np.arctan2(float(ty) - float(cy), float(tx) - float(cx))
         in_sandtrap = is_in_sand_trap(
             current_point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap)
-        splash_zone_poly_points = splash_zone(float(distance), float(
-            angle), float(conf), self.skill, current_point, in_sandtrap)
-        return self.shapely_poly.contains(ShapelyPolygon(splash_zone_poly_points))
 
-    def splash_zone_within_sand(self, current_point: Tuple[float, float], target_point: Tuple[float, float], conf: float) -> bool:
-        if type(current_point) == Point2D:
-            current_point = tuple(Point2D)
+        target_sandtrap = is_in_sand_trap(
+            target_point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap)
 
-        if type(target_point) == Point2D:
-            target_point = tuple(Point2D)
-
-        distance = np.linalg.norm(np.array(current_point).astype(
-            float) - np.array(target_point).astype(float))
-        cx, cy = current_point
-        tx, ty = target_point
-        angle = np.arctan2(float(ty) - float(cy), float(tx) - float(cx))
-        in_sandtrap = is_in_sand_trap(
-            current_point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap)
         splash_zone_poly_points = splash_zone(float(distance), float(
-            angle), float(conf), self.skill, current_point, in_sandtrap)
+            angle), float(conf), self.skill, current_point, in_sandtrap, target_sandtrap)
+
+        splash_in_poly = self.shapely_poly.contains(ShapelyPolygon(splash_zone_poly_points))
+
+        splash_in_sand = False
```

Comment:
A further optimization could be that if `splash_in_poly == True`, directly return. There's no need to compute `splash_in_sand` anymore, since we know this target_point will be discarded by the caller and not added to the queue during A* search.

Reply:
Tried this out, but seemed to make the algorithm run slower... Not exactly sure why, discarding this idea for now.